dev/image-guidelines.md

```diff
@@ -0,0 +1,329 @@
+
+## Excalidraw object library
+
+This file is an Excalidraw library containing reusable shape objects for our drawings.
+
+![library.excalidrawlib](library.excalidrawlib)
+
+If you find or create objects that you think will be useful in future drawings then please add them to this library, save the changes and commit the updated version back to the repo.
+
+
+## Screenshots
+
+### Screen size
+
+While we may do our daily work on screens with high display resolutions,
+the documentation will likely be read on smaller displays or possibly even printed.
+It needs to be legible on typical screens without a magnifying glass or zooming in.
+
+Ideally we want screenshots in the documentation to have a comfortably low resolution.
+This can be achieved by either:
+
+- only showing the part of the screen that the user needs to see.
+- using a constrained screen size for full-screen screenshots.
+
+These are explained below.
+
+#### Partial screens
+
+Screenshots that show only part of the screen area are relatively straightforward.
+
+![](image-guidelines/screen_partial.png)
+
+A useful trick in Chrome Dev Tools is that you can right-click on a DOM element and capture an image of just that element.
+![](image-guidelines/node_screenshot.png)
+
+#### Full screens
+
+When full-size screenshots are needed we can improve their legibility by simulating a smaller screen size.
+The "dev tools" of modern browsers allows you to simulate arbitrary screen sizes, including many common phone screens.
+Velociraptor's GUI is not a fully responsive web application but it's layout remains usable at quite small screen resolutions.
+
+For screenshots the following 2 sizes seem to work reasonably well:
+
+- 1280x960 (first choice)
+- 1600x1200 (sometimes needed to show wide table content)
+
+![](image-guidelines/screen_size.png)
+*These particular constrained sizes can be configured as presets in Chrome's "dev tools" for quick access.*
+
+### Velociraptor theme
+
+For consistency in the documentation we should use a single GUI theme for all screenshots
+:   the Velociraptor Light theme which is the default for new installations.
+
+#### Theme tweaks
+
+While screenshots do need to depict the actual application UI, it's reasonable to tweak some aspects to make the screenshots a bit clearer for documentation purposes.
+These may include:
+1. using a slightly heavier font weight: 500 rather than 400. On Windows the fonts seem to render slightly lighter for some unknown reason.
+2. using pure black or white as the text color instead of lower contrast tones.
+3. reducing the opacity of the modal backdrop so that items in the background are not excessively dimmed
+4. deleting distracting elements from the DOM if necessary.
+5. adding a glow effect via CSS to emphasize a specific element.
+
+Points 1-3 can be patched into the GUI's CSS and compiled to make a documentation-specific binary. A patch is provided in this folder named `documentation_tweaks_{commit_id}.patch`
```

Comment:
Can we just add this as another theme?

Reply:
We could but then it probably shouldn't be one that's visible/selectable for users. If there's some way to hide it in the GUI user prefs but still allow it to be set it via user_options() that would be ideal. It seems like overkill though.

Since those 2 CSS files hardly ever change it's easy to just patch it when needing to make screenshots. And updating the patch on rare occasions isn't difficult.

Reply:
I dont mind it being selectable - that makes it easier for people to use. I dont think it is acceptable to patch the code just for docs - this is likely to result in accidents like commiting this to the main branch with another PR accidentally.

Reply:
Ok. Thanks for adding it: https://github.com/Velocidex/velociraptor/pull/3764